frontend/src/employee-frontend/components/absences/StaffAttendance.tsx

```diff
@@ -144,48 +148,56 @@ const StaffAttendanceRow = memo(function StaffAttendanceRow({
   )
 })
 
-interface StaffAttendanceCellProps {
-  attendance: StaffAttendance
-  updateAttendances: (attendance: StaffAttendance) => Promise<() => void>
-  disabled: boolean
-}
-
 const DisabledStaffIcon = styled(FontAwesomeIcon)`
   font-size: 15px;
   color: ${colors.greyscale.dark};
 `
 
+const InactiveCell = ({ date }: { date: LocalDate }) => {
+  const { i18n } = useTranslation()
+  return (
+    <div className={'absence-cell disabled-staff-cell-container'}>
+      <Tooltip
+        tooltipId={`tooltip_disabled-staff-cell-${date.formatIso()}`}
+        tooltipText={i18n.absences.table.disabledStaffCellTooltip}
+        place={'top'}
+      >
+        <DisabledStaffIcon icon={faTimes} />
+      </Tooltip>
+    </div>
+  )
+}
+
+interface StaffAttendanceCellProps {
+  attendance: GroupStaffAttendance | undefined
+  updateAttendance: (count: number) => Promise<void>
+}
+
 const StaffAttendanceCell = memo(function StaffAttendanceCell({
   attendance,
-  updateAttendances,
-  disabled
+  updateAttendance
 }: StaffAttendanceCellProps) {
-  const [inputValue, setInputValue] = useState(formatDecimal(attendance.count))
+  const initial: number | undefined = attendance?.count
+  const [inputValue, setInputValue] = useState(formatDecimal(initial))
   const [updatedValue, setUpdatedValue] = useState<number>()
   const [editing, setEditing] = useState(false)
-  const { i18n } = useTranslation()
 
   useEffect(() => {
     if (!editing) {
-      setInputValue(formatDecimal(attendance.count))
+      setInputValue(formatDecimal(initial))
```

Comment:
Since you did state management refactoring as a part of this commit anyway, it would be nice if you could refactor this whole component's state management since it's a bit of a mess.

Reply:
Done! It was a mess 😄